src/software/ai/hl/stp/tactic/crease_defender/crease_defender_fsm.h

```diff
@@ -70,14 +92,25 @@ struct CreaseDefenderFSM
         DEFINE_SML_STATE(MoveFSM)
         DEFINE_SML_EVENT(Update)
         DEFINE_SML_SUB_FSM_UPDATE_ACTION(blockThreat, MoveFSM)
+        DEFINE_SML_STATE(DribbleFSM)
+        DEFINE_SML_GUARD(ballNearbyWithoutThreat)
+        DEFINE_SML_SUB_FSM_UPDATE_ACTION(prepareGetPossession, DribbleFSM)
 
         return make_transition_table(
             // src_state + event [guard] / action = dest_state
-            *MoveFSM_S + Update_E / blockThreat_A, MoveFSM_S = X,
+            *MoveFSM_S + Update_E[ballNearbyWithoutThreat_G] / prepareGetPossession_A =
+                DribbleFSM_S,
+            MoveFSM_S + Update_E / blockThreat_A, MoveFSM_S = X,
+            DribbleFSM_S + Update_E[!ballNearbyWithoutThreat_G] / blockThreat_A =
+                MoveFSM_S,
```

Comment:
You should add an internal transition (from DribbleFSM_S to DribbleFSM_S) here:

```suggestion
            DribbleFSM_S + Update_E[!ballNearbyWithoutThreat_G] / blockThreat_A =
                MoveFSM_S,
            DribbleFSM_S + Update_E / prepareGetPossession_A,
```

and remove this:

https://github.com/UBC-Thunderbots/Software/blob/5bd755beb03a51d138507c0e4cc3b075af5b09ef/src/software/ai/hl/stp/tactic/crease_defender/crease_defender_tactic.cpp#L62-L63

What you doing is calling `process_event` with `DribbleFSM::Update`, which runs the `DribbleFSM` sub-FSM no matter what state `CreaseDefenderFSM` is in.

The reason why your code failed before you added this extra `process_event` is because when the `process_event` was called with `CreaseDefenderFSM::Update` and the `CreaseDefenderFSM` was in `DribbleFSM_S`, if `ballNearbyWithoutThreat_G` was true then nothing would happen. We need always need to execute some action that yields a primitive, so adding an internal transition for `DribbleFSM_S` will resolve this issue.